### PR TITLE
Post exp: update Table tBody Dynamically

### PR DESCRIPTION
**Pull request by jinnatul**
**What's New**

- Change Table Body Dynamically
- initiated table length phy exp
- button add each exp phy and update db experament
- get click table ID and Current Table length
- find baseTBody for requested table
- implemented add new row

Screenshot from 2019-11-28 01-16-32
Screenshot from 2019-11-28 01-16-39
Screenshot from 2019-11-28 01-16-47

### CODE REVIEW

File: src/wwwroot/api/scripts/GetExperimentsOfPhysics.js [hunk 1]
```diff
@@ -6,25 +9,67 @@ $(document).ready(function () {
...
+
+        clickedTableCurrentLength++;
+        
+        tBody = tBody.replace(/"group"/g, "'group"+ clickedTableCurrentLength + "'");
+        tBody = tBody.replace(/"typeOf"/g, "'typeOf"+ clickedTableCurrentLength + "'");
+        
+        initialLengthOfPhyExp[clickedTableID][clickedTableID] = clickedTableCurrentLength;
+        
+        //console.log(tBody);
+        $('#'+clickedTableID).append('<tr>' + tBody + '</tr>');
```
Comment (shunjid) [inline src/wwwroot/api/scripts/GetExperimentsOfPhysics.js]: This line has a minor bug. It's adding unnecessary extra **table row** when there is any rowspan in tBody. **I'll fix this.**

File: src/wwwroot/api/scripts/GetExperimentsOfPhysics.js [hunk 2]
```diff
@@ -37,74 +82,104 @@ function setHeaders(list, expTable) {
...
+        }
+        else {
 
-                if(val === 0){
-                    radioString += '<label><input type="radio" name="group'+i+'" checked/> <span>বায়ু মন্ডলের চাপ</span> </label>' + '<br>';
-                    radioString += '<label><input type="radio" name="group'+i+'"/> <span>বায়ু মন্ডলের চাপের বেশী</span> </label>' + '<br>';
+            let value = exp[i][columns[0]];
+            let radioString = "";
+            //চাপ
+            if(columns[0] === "চাপ") {
```
Comment (shunjid) [inline src/wwwroot/api/scripts/GetExperimentsOfPhysics.js]: Needs to make it dynamic. **Putting it into future factory.**

File: src/wwwroot/api/scripts/GetExperimentsOfPhysics.js [hunk 3]
```diff
@@ -37,74 +82,104 @@ function setHeaders(list, expTable) {
...
-            if(val.length > 0) {
-                let subRow = $("<tr/>");
-                val.forEach(element => {
-                    subRow.append($('<tr/>').html(element)); 
-                });
-                row.append($('<td/>').html(subRow));
-                continue;
+            for (let colIndex = 0; colIndex < columns.length; colIndex++){
+                
+                if((columns[0] === "ব্যবস্থা" || columns[0] ==="চাপ") && (colIndex === 0)) continue;
```
Comment (shunjid) [inline src/wwwroot/api/scripts/GetExperimentsOfPhysics.js]: **future.factory (+2)**

File: src/wwwroot/api/scripts/GetExperimentsOfPhysics.js [hunk 4]
```diff
@@ -37,74 +82,104 @@ function setHeaders(list, expTable) {
...
+                rowStr += "<tr>";
+                for (let colIndex = 0; colIndex < columns.length; colIndex++){
+                    let eachVal = exp[i][columns[colIndex]];
+                    if(eachVal.length > 1){
+                        rowStr += "<td>"+eachVal[posCnt]+"</td>";
+                    }
+                }
+                posCnt++;
+                rowStr += "</tr>";
+            }
```
Comment (shunjid) [inline src/wwwroot/api/scripts/GetExperimentsOfPhysics.js]: > This is smarter. Bravo.

File: src/wwwroot/api/scripts/GetExperimentsOfPhysics.js (hunk 1 above)
Comment (jinnatul) [reply src/wwwroot/api/scripts/GetExperimentsOfPhysics.js]: oh, shit !!!

File: src/wwwroot/api/scripts/GetExperimentsOfPhysics.js (hunk 2 above)
Comment (jinnatul) [reply src/wwwroot/api/scripts/GetExperimentsOfPhysics.js]: I need to some time's then I hope, I can handle this dynamically

**Pull request merged**